### DISCUSSION TITLE
Namespacing Fozzie component specific Vuex stores internally

### DISCUSSION THREAD

**Discussion by kevinrodrigues** [justeattakeaway/fozzie-components#730] [General]
Just wanted to get some thoughts/opinions on an issue i've come across with Vuex registration.

I have a component that could be used in multiple places/verticals. It registers a Vuex store name as `xyz` internally. If the consuming app happens to register the same name (`xyz`) this could break the components Vuex store on the consuming apps side (silently in some cases). Have we considered a naming convention for fozzie-component related Vuex registration? I did looked at the option of exposing the store from within the component so the consuming app can register it themselves, possibly avoiding a name clash, but for a component that is intended to be used in multiple areas of JET having to set the store up manually in each consuming app feels unnecessary & I would like to avoid this if possible (if it was used in one place it would be ok) 

## Problem
```javascript
// [FOZZIE-COMPONENT]: we register a Vuex store...
```
![carbon (6)](https://user-images.githubusercontent.com/2299779/108509913-c5e99a00-72b5-11eb-8188-e94e47f33081.png)

```javascript
// [CONSUMING-APP]: We might already have a store being registered with the same name...
```
![carbon (5)](https://user-images.githubusercontent.com/2299779/108508629-ffb9a100-72b3-11eb-9219-23b973427924.png)

``` javascript
// [OUTPUT]: This causes the following error

unknown local action type: actionXYZ, global type: XYZ/foo
```

**Comment by jamieomaguire** [comment]
So would following an `f-module-name` or  `f-component-module-name` convention for internal modules like we already do for components resolve this issue?

**Reply by kevinrodrigues** [reply]
Yeah that would work. I was initially thinking something like `fComponentModule` as the store name.

**Comment by kevinrodrigues** [comment]
I've gone with the following naming convention in case anyone needs to do the same:

`fsearchboxModule`

format: `[f][componentName]Module`
